builder/build.go

```diff
@@ -649,6 +650,13 @@ func Build(pkgName, outpath, tmpdir string, config *compileopts.Config) (BuildRe
 	result.Binary = result.Executable // final file
 	ldflags := append(config.LDFlags(), "-o", result.Executable)
 
+	if config.Options.BuildMode == "c-shared" {
+		if config.GOOS() != "wasip1" {
```

Comment:
I think we’ll quickly relax this, perhaps better to check the triple or GOARCH for a `wasm` prefix?

Reply:
I’m fairly certain that @deadprogram will want to use this for `unknown`.

Reply:
Yes. But to avoid too much complexity in this PR I've left it out. I think I tried to do it all at once but it became a bit more complex than I liked. Better to get it working for wasip1 first and then expand support to other targets (which should be relatively easy).